hironx_ros_bridge/src/hironx_ros_bridge/hironx_client.py

```diff
@@ -667,13 +667,20 @@ def checkEncoders(self, jname='all', option=''):
             self.rh_svc.power('all', SWITCH_OFF)
             return 0
 
+        is_result_hw = True
         print self.configurator_name, 'calib-joint ' + jname + ' ' + option
-        self.rh_svc.initializeJointAngle(jname, option)
+        is_result_hw = is_result_hw and self.rh_svc.initializeJointAngle(jname, option)
         print self.configurator_name, 'done'
-        self.rh_svc.power('all', SWITCH_OFF)
+        is_result_hw = is_result_hw and self.rh_svc.power('all', SWITCH_OFF)
         self.goActual()  # This needs to happen before turning servo on.
         time.sleep(0.1)
-        self.rh_svc.servo(jname, SWITCH_ON)
+        is_result_hw = is_result_hw and self.rh_svc.servo(jname, SWITCH_ON)
```

Comment:
@130s You should be able to check the calibration state. Please refer to the function isCalibDone. The servoState variable in class RobotStatus (refer to /opt/ros/indigo/share/hrpsys/idl/RobotHardwareService.idl) also holds the calibration state. So doing
>>> rstt = self.rh_svc.getStatus()
and checking rstt.servoState[0][0] will return the calibration state of joint 0.